### ISSUE TITLE
Failed installation, zendframework/zend-barcode version error

### ISSUE THREAD

**Issue by watista** [tig-nl/postnl-magento2#145]
## Please follow the guide below

- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like.
- We may ask some questions or ask you to provide addition information after you placed your request.

---

### Make sure you are using the *latest* version: https://tig.nl/postnl-magento-extensies/
Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running the latest version of the TIG PostNL Magento extension.

---

### What is the purpose of your *issue*?
- [ ] Feature request (request for a new functionality)
- [x ] Bug report (encountered problems with the TIG PostNL Magento extension)
- [ ] Extension support request (request for adding support for a new extension)
- [ ] Other

---

### If your issue contains security sensitive information
** If your issue may contain sensitive information, please send us an e-mail at security@tig.nl.**
- [x ] This issue does not contain any security sensitive information

---

### If the purpose of your issue is a *feature request*
- As a *..certain type of user..* I’d like to *..describe what you want..* in order to *..write goal(s)..*
- Suggestions and background: 
  If you have ideas for the solution write them down here. Include images, a picture is more worth than 1,000 words. If none, remove this line.

### If the purpose of your issue is a *bug report*
**Please describe the precise steps to reproduce your issue:**
- Step 1, Step 2, ...
I tried installing the Postnl-magento2 module but i get an error that another version of zendframework/zend-barcode is locked. Also tried updating composer beforehand. Same result. I use magento 2.3.1
Any way of using another zend-barcode version?
- Expected result
Succesfull installation

- Actual result (Please provide screenshots if applicable)

Composer installation error message
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install tig/postnl-magento2 1.8.1
    - Conclusion: don't install tig/postnl-magento2 1.8.0.1
    - Conclusion: remove zendframework/zend-barcode 2.8.0
    - Installation request for tig/postnl-magento2 ^1.8 -> satisfiable by tig/postnl-magento2[1.8.0, 1.8.0.1, 1.8.1].
    - Conclusion: don't install zendframework/zend-barcode 2.8.0
    - tig/postnl-magento2 1.8.0 requires zendframework/zend-barcode 2.5.2|2.7 -> satisfiable by zendframework/zend-barcode[2.5.2, 2.7.0].
    - Can only install one of: zendframework/zend-barcode[2.7.0, 2.8.0].
    - Can only install one of: zendframework/zend-barcode[2.5.2, 2.8.0].
    - Installation request for zendframework/zend-barcode (locked at 2.8.0) -> satisfiable by zendframework/zend-barcode[2.8.0].

Installation failed, reverting ./composer.json to its original content.
```
- Workaround (If available, otherwise just write "None")
None

### If the purpose of your issue is about anything else, please describe your issue here
- Explain your issue here

### TIG support

- Please ask your question in English to ensure that your issue can help other people internationally. Nevertheless we will respond in English.
- If you want support with installing the PostNL extension, please contact citservicedesk@postnl.nl for free support, or contact us directly at support@tig.nl for paid support plans.

**Comment by tig-dennisvanderhammen** [comment]
Hi @watista,

Thank you for pointing this out. We are aware of this issue and are currently testing if everything works okay with zend-barcode 2.8.0. If we can't find any issues we'll add this in the upcoming release.

**Comment by rickschippers** [comment]
I had similar issues. There is a workaround to get it installed / working.

zend-barcode is required by temando/shipping-m2 which is installed by default in magento2.
It's constraint is set to:  "zendframework/zend-barcode": "^2.7.0". So it works fine with a 2.7 version of zend-barcode. 

But if you have 2.3.3 already installed, you'll have a 2.8.x version of zend-barcode and this version is in your composer.lock file as well.

If you manually add:  "tig/postnl-magento2": "~1.8.1" to composer.json and then run composer update, it will install fine. zendframework/zend-barcode will downgrade to 2.7.1 and tig/postnl-magento2 1.8.1 will install just fine.

**Comment by tig-svenluiks** [comment]
Hey Watista,

Could you let us know if the answer my colleague gave you helped you solve the problem?

Looking forward to you response.

Greetings,

Sven

**Comment by watista** [comment]
Hi Sven,

Assuming Rick Schippers is your colleague (since Dennis didn't gave a solution or fix).

I added: "tig/postnl-magento2": "~1.8.1" to my composer.json file, updated composer and it did install. Storefront and admin are working except for viewing my shopping cart or checking out.
The GET request for the cart/ page is giving a 500 internal server error. 

The nginx error log:
```2020/01/13 16:42:56 [error] 529#529: *313 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught TypeError: Return value of Magento\Framework\Encryption\Adapter\SodiumChachaIetf::decrypt() must be of the type string, boolean returned in /var/www/dev/vendor/magento/framework/Encryption/Adapter/SodiumChachaIetf.php:68
Stack trace:
#0 /var/www/dev/vendor/magento/framework/Encryption/Encryptor.php(373): Magento\Framework\Encryption\Adapter\SodiumChachaIetf->decrypt('[encryption-key...]')
#1 /var/www/dev/vendor/tig/postnl-magento2/Config/Provider/AccountConfiguration.php(103): Magento\Framework\Encryption\Encryptor->decrypt('[encryption-key...]')
#2 /var/www/dev/vendor/tig/postnl-magento2/Config/CheckoutConfiguration/IsShippingOptionsActive.php(144): TIG\PostNL\Config\Provider\AccountConfiguration->getApiKey()
#3 /var/www/dev/vendor/tig/postnl-magento2/Config/CheckoutConfiguration/IsShippingOptionsActive.php(101): TIG\PostNL\Config\CheckoutConfiguration\IsShippingOptionsActive->hasValidApiSettings()
#4 /var/www/dev/vendor/tig/postnl-magento2/Config/P" while reading response header from upstream, client: 10.1.0.1, server: [server], request: "GET /checkout/cart/ HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.2-fpm.sock:", host: "[host]", referrer: "[referrer]" 
```

There are no errors shown in de system or debug log of magento2.

@rickschippers Did you also encounter this problem? If yes, how did you manage to fix it?
@tig-svenluiks & @tig-dennisvanderhammen Is there an ETA for the next update where this bug will be fixed? 

Thanks in advance!

**Comment by watista** [comment]
Sorry for the previous comment, the error i got was because of a wrong server migration, the crypt key were invalid. 

After fixing this the cart and checking out are also working (storefront and backend were already working). This means that the fix given by @rickschippers is working. In my composer output i saw zendframework/zend-barcode downgrading to 2.7.

@tig-svenluiks / @tig-dennisvanderhammen Do you have an ETA for the next update? I would rather install the module the proper way on my live environment.

**Comment by tig-dennisvanderhammen** [comment]
Hi @watista,

We have fixed this issue in v1.9.0. The postnl-magento2 extension accepts more zend-barcode versions now!
